toopher/__init__.py

```diff
@@ -110,106 +122,77 @@ def validate(self, data, request_token=None, ttl=DEFAULT_IFRAME_TTL):
 
         return data
 
-    def signature(self, data):
+    def _signature(self, data):
         to_sign = urllib.urlencode(sorted(data.items())).encode('utf-8')
         secret = self.client.client_secret.encode('utf-8')
         return base64.b64encode(hmac.new(secret, to_sign, hashlib.sha1).digest())
 
-    def get_oauth_uri(self, uri, params, ttl):
+    def _get_oauth_signed_url(self, uri, params, ttl):
         params['expires'] = str(int(time.time()) + ttl)
         return self.client.sign(uri + '?' + urllib.urlencode(params))[0]
 
 
 class ToopherApi(object):
     def __init__(self, key, secret, api_url=None):
-        self.client = requests_oauthlib.OAuth1Session(key, client_secret=secret)
-        self.client.verify = True
-
-        base_url = api_url if api_url else DEFAULT_BASE_URL
-        self.base_url = base_url.rstrip('/')
-
-    def pair(self, pairing_phrase, user_name, **kwargs):
-        uri = self.base_url + "/pairings/create"
-        params = {'pairing_phrase': pairing_phrase,
-                  'user_name': user_name}
+        self.advanced = AdvancedApiUsageFactory(key, secret, api_url)
 
+    def pair(self, username, phrase_or_num=None, **kwargs):
+        params = {'user_name': username}
         params.update(kwargs)
+        if phrase_or_num:
+            if any(c.isdigit() for c in phrase_or_num):
+                url = '/pairings/create/sms'
+                params.update(phone_number=phrase_or_num)
```

Comment:
Either that or we could check for phone number patterns like `xxx-xxx-xxxx` or `(xxx) xxx-xxxx` or `xxxxxxxxxx` etc. But since none of our pairing phrases have been numbers and we control that generation, and there are many different ways to type a phone number, we thought this might be a simpler, cleaner way to look for that.